src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java

```diff
@@ -423,9 +434,34 @@ public InnReachResponseDTO finalCheckIn(String trackingId, String centralCode, B
       .ifPresent(this::removeHoldingsTransactionInfo);
     clearPatronAndItemInfo(transaction.getHold());
 
+    executeDeleteVirtualRecordsWithDelay(transaction.getHold());
+
     return success();
   }
 
+  private void executeDeleteVirtualRecordsWithDelay(TransactionHold hold) {
+
+    var folioItemId = hold.getFolioItemId();
+    var folioHoldingId = hold.getFolioHoldingId();
+    var folioInstanceId = hold.getFolioInstanceId();
+    var folioLoanId = hold.getFolioLoanId();
+
+    var configDataList=
+            configurationService.fetchConfigurationsDetailsByModule(CHECKOUT);
+
+    Long checkOutTimeDuration = getCheckOutTimeDuration(configDataList.getResult());
+
+    log.info("Checkout Time Duration is : {}", checkOutTimeDuration);
+
+    log.info("deleteVirtualRecords execution started at : " + new Date());
+    var task = new FolioAsyncExecutorWrapper(folioExecutionContext,
+            () -> virtualRecordService.deleteVirtualRecords(folioItemId, folioHoldingId,
+                    folioInstanceId, folioLoanId));
+
+    if(taskExecutor.schedule(task, new Date(System.currentTimeMillis() + checkOutTimeDuration)).isDone())
```

Comment:
this method from the Future will always return FALSE.

Reply:
Do all the logging inside the `deleteVirtualRecords` method

Reply:
Please be careful, you must schedule the task, but it is unnecessary to check the result here and add a logging message. But you deleted the complete statement with the scheduling part.

Reply:
yes.. understood